fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/insert/InsertUtils.java

```diff
@@ -345,13 +347,17 @@ private static Expression castValue(Expression value, DataType targetType) {
      * get target table from names.
      */
     public static TableIf getTargetTable(Plan plan, ConnectContext ctx) {
-        if (!(plan instanceof UnboundTableSink)) {
-            throw new AnalysisException("the root of plan should be UnboundTableSink"
-                    + " but it is " + plan.getType());
+        if (plan instanceof UnboundTableSink) {
```

Comment:
What is the difference between this 2 branch?
We need to try best to avoid `if else` for different type of table

Reply:
UnboundTableSink is too complex, so we should refactor to UnboundTableSink first. 
Before the refactor, to simplify the problem，we should use 'if-else' to distinguish olap and hive for the moment.
。